src/app/models/workspace.server.model.ts

```diff
@@ -70,6 +70,22 @@ const compileWorkspaceModel = (db: Mongoose): IWorkspaceModel => {
     return this.findOneAndUpdate({ _id: workspaceId }, { title }, { new: true })
   }
 
+  WorkspaceSchema.statics.deleteWorkspace = async function ({
+    workspaceId,
+    session,
+  }: {
+    workspaceId: IWorkspaceSchema['_id']
+    session?: ClientSession
+  }) {
+    const deleted = await this.deleteOne(
```

Comment:
why not use `findOneAndRemove` so we can return the deleted document, which is more informative for the caller?